demo/client.rb

```diff
@@ -0,0 +1,57 @@
+# frozen_string_literal: true
+
+# This is generated code!
+
+module Weather
+  class Client < Smithy::Client::Base
+    self.service_shape = Shapes::WEATHER
+
+    add_plugin(Smithy::Client::Plugins::NetHTTP)
+
+
+    # TODO
+    def initialize(*args)
+      super(*args)
+    end
+
+    # TODO!
+    def get_city(params = {}, options = {})
+      input = build_input(:get_city, "example.weather#GetCity", params)
```

Comment:
I was looking how build inputs param for `operation_name` are used in plugins and handlers. it seems like we always access these through the context object. Since the context object will have operation name and operation shape (operation.api) so I think the check will be fine?   I might be lookin at the wrong places so let me know.

I also think OperationShape should have an instance method that provides rubyified method name. But the structure of operations for service shape should follow shape id.

Reply:
Per team discussion, we are going to go with symbolized ruby names as the key here since the operation shape will have the shape id as one of their shape attributes